cmd/cluster/none/create.go

```diff
@@ -60,39 +60,46 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 	// - NodeExternalIP
 	// - NodeInternalIP
 	if opts.NonePlatform.APIServerAddress == "" {
-		kubeClient := kubeclient.NewForConfigOrDie(util.GetConfigOrDie())
-		nodes, err := kubeClient.CoreV1().Nodes().List(ctx, v1.ListOptions{Limit: 1})
-		if err != nil {
-			return fmt.Errorf("unable to fetch node objects: %w", err)
+		if opts.NonePlatform.APIServerAddress, err = GetAPIServerAddress(ctx); err != nil {
+			return err
 		}
-		if len(nodes.Items) < 1 {
-			return fmt.Errorf("no node objects found: %w", err)
-		}
-		addresses := map[corev1.NodeAddressType]string{}
-		for _, address := range nodes.Items[0].Status.Addresses {
-			addresses[address.Type] = address.Address
-		}
-		for _, addrType := range []corev1.NodeAddressType{corev1.NodeExternalDNS, corev1.NodeExternalIP, corev1.NodeInternalIP} {
-			if address, exists := addresses[addrType]; exists {
-				opts.NonePlatform.APIServerAddress = address
-				break
-			}
-		}
-		if opts.NonePlatform.APIServerAddress == "" {
-			return fmt.Errorf("node %q does not expose any IP addresses, this should not be possible", nodes.Items[0].Name)
-		}
-		log.Info(fmt.Sprintf("detected %q from node %q as external-api-server-address", opts.NonePlatform.APIServerAddress, nodes.Items[0].Name))
 	}
 
+	SetCommon(exampleOptions, opts)
+
+	exampleOptions.None = &apifixtures.ExampleNoneOptions{
+		APIServerAddress: opts.NonePlatform.APIServerAddress,
+	}
+	return nil
+}
+
+func SetCommon(exampleOptions *apifixtures.ExampleOptions, opts *core.CreateOptions) {
```

Comment:
For my taste `SetCommon` does not look right. It is too unspecific and small to justify code-reuse and makes it harder to get which commands are defaulted. What do you think?

Reply:
Removed it